src/main/java/bio/terra/service/duos/DuosDao.java

```diff
@@ -39,7 +48,7 @@ String getTdrServiceAccountEmail() {
   }
 
   @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE)
-  public DuosFirecloudGroupModel insertAndRetrieveFirecloudGroup(
+  public UUID insertFirecloudGroup(
```

Comment:
I originally thought that we'd store the DUOS ID of a linked DUOS dataset in the `snapshot` table.  But since we are storing our `duos_firecloud_group` surrogate key instead (which is autogenerated), I changed row insertion to return this generated key so that we have access to it.  This will prove useful when I build the flight to link a snapshot to a DUOS dataset.